### PR TITLE
feat(skill-loader): recursive merge for multi-file skills

### PR DESCRIPTION
**Pull request by codewithkenzo**
## Summary

Extends the skill loader to recursively merge all `.md` files from skill subdirectories into the skill template. This enables complex multi-file skills like Vercel's `react-best-practices` to work without pre-compilation.

## Changes

- **New `collectMdFilesRecursive()` helper** - Recursively collects `.md` files from subdirectories up to depth 3
- **Modified `loadSkillFromPath()`** - Integrates recursive merge, appends merged content after SKILL.md body
- **Test fixtures** - 7 fixture skills covering edge cases

## Features

- Strips frontmatter from subdir files (only SKILL.md frontmatter has semantic meaning)
- Excludes hidden files, symlinks, root README.md/LICENSE.md
- Sorts merged files alphabetically by path
- Backward compatible: single-file skills unchanged

## Testing

- **14 new tests** added, all 60 tests pass
- **Tested with Vercel react-best-practices** (55 files, 71KB merged successfully)

## Files Changed

- `src/features/opencode-skill-loader/loader.ts` - Implementation
- `src/features/opencode-skill-loader/loader.test.ts` - Tests
- `src/features/opencode-skill-loader/__fixtures__/**` - Test fixtures (7 skills)

**20 files changed** (well under Greptile's 500 file limit)

---
## Summary by cubic
Enables multi-file skills by recursively merging .md files from subdirectories into the skill template. Content is appended after SKILL.md; single-file skills remain unchanged.

- **New Features**
  - Recurses up to depth 3 and sorts merged files by path.
  - Skips hidden files, symlinks, and root README.md/LICENSE.md.
  - Strips frontmatter from subdir files; only SKILL.md frontmatter is used.
  - Adds a merge marker comment and integrates into loadSkillFromPath.
  - 14 tests and 7 fixtures added; verified with Vercel react-best-practices.

<sup>Written for commit f5b3842d24c283b22b0f6b306e0e55d03f79e6ff. Summary will update on new commits.</sup>

### CODE REVIEW

File: src/features/opencode-skill-loader/loader.ts [hunk 1]
```diff
@@ -68,6 +106,12 @@ async function loadSkillFromPath(
     const mcpJsonMcp = await loadMcpJsonFromDir(resolvedPath)
     const mcpConfig = mcpJsonMcp || frontmatterMcp
 
+    const subdirFiles = await collectMdFilesRecursive(resolvedPath, 0, 3, '')
```
[the remark this answers is not in the document]
Comment (codewithkenzo) [reply src/features/opencode-skill-loader/loader.ts:109]: This is a false positive.

`collectMdFilesRecursive(resolvedPath, ...)` only scans **within that specific skill's directory** (`resolvedPath`). It merges `.md` files from subdirectories of the same skill (e.g., `react-best-practices/rules/*.md`).

It does NOT merge across different skills - each skill is isolated to its own directory.

The `single-file-skill` test fixture explicitly verifies this: a skill with no subdirs gets no merged content and no merge comment.

**Comment by codewithkenzo** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by codewithkenzo** [comment]
Regarding cubic's concern about cross-skill contamination: This is a **false positive**.

`collectMdFilesRecursive(resolvedPath, ...)` only scans **within that specific skill's own directory**. Each skill is completely isolated - it merges `.md` files from its own subdirectories (e.g., `react-best-practices/rules/*.md`), not from other skills.

The `single-file-skill` test fixture explicitly verifies this: a skill with no subdirs gets no merged content and no merge comment.

See inline reply for details.
